### PR TITLE
Updated Löve standard to 11.1

### PR DESCRIPTION
**Pull request by giann**
See https://love2d.org/wiki/11.0

### CODE REVIEW

**Comment by mpeterv** [comment]
Thank you! Just in case, @Positive07 could you take a look at this?

File: src/luacheck/love_standard.lua [hunk 1]
```diff
@@ -109,7 +119,8 @@ local love = {
          "getFullscreenModes","getIcon","getMode","getPixelScale","getPosition","getTitle",
          "hasFocus","hasMouseFocus","isDisplaySleepEnabled","isMaximized","isOpen","isVisible",
          "maximize","minimize","requestAttention","setDisplaySleepEnabled","setFullscreen",
-         "setIcon","setMode","setPosition","setTitle","showMessageBox","toPixels")
+         "setIcon","setMode","setPosition","setTitle","showMessageBox","toPixels",
+         "updateMode","isMinimized","restore")
```
Comment (Positive07) [inline src/luacheck/love_standard.lua]: Missing field `getDPIScale`

File: src/luacheck/love_standard.lua [hunk 2]
```diff
@@ -16,6 +16,7 @@ local love = {
       gamepadpressed = read_write,
```
Comment (Positive07) [inline src/luacheck/love_standard.lua]: Missing callback `errorhandler`

File: src/luacheck/love_standard.lua [hunk 3]
```diff
@@ -68,7 +74,11 @@ local love = {
...
          "setBlendMode","setCanvas","setColor","setColorMask","setDefaultFilter","setFont",
          "setLineJoin","setLineStyle","setLineWidth","setNewFont","setShader","setPointSize",
-         "setScissor","setStencilTest","setWireframe","shear","stencil","translate"),
+         "setScissor","setStencilTest","setWireframe","shear","stencil","translate",
+         "captureScreenshot","getImageFormats","newArrayImage","newVolumeImage","newCubeImage",
+         "getTextureTypes","drawLayer","setDepthMode","setMeshCullMode","setFrontFaceWinding",
+         "applyTransform","replaceTransform","transformPoint","inverseTransformPoint","getStackDepth",
+         "flushBatch","validateShader"),
 
       image = standards.def_fields("isCompressed","newCompressedData","newImageData"),
```
Comment (Positive07) [inline src/luacheck/love_standard.lua]: Missing field `newCubeFaces` (this function isn't documented in the wiki yet)

File: src/luacheck/love_standard.lua [hunk 4]
```diff
@@ -48,14 +50,18 @@ local love = {
...
       event = standards.def_fields("clear","poll","pump","push","quit","wait"),
 
       filesystem = standards.def_fields("append","areSymlinksEnabled","createDirectory","exists",
          "getAppdataDirectory","getDirectoryItems","getIdentity","getLastModified",
          "getRealDirectory","getRequirePath","getSaveDirectory","getSize","getSource",
          "getSourceBaseDirectory","getUserDirectory","getWorkingDirectory","init","isDirectory",
          "isFile","isFused","isSymlink","lines","load","mount","newFile","newFileData","read",
-         "remove","setIdentity","setRequirePath","setSource","setSymlinksEnabled","unmount","write"),
+         "remove","setIdentity","setRequirePath","setSource","setSymlinksEnabled","unmount","write",
+         "getInfo","setCRequirePath","getCRequirePath"),
```
Comment (Positive07) [inline src/luacheck/love_standard.lua]: I noticed reviewing this that the `love.font` module is missing which should have the following fields:
```lua
font = standards.def_fields("newImageRasterizer", "newGlyphData", "newTrueTypeRasterizer", "newBMFontRasterizer", "newRasterizer"),
```

**Comment by Positive07** [comment]
That's a lot of stuff... there is also other "not so official" stuff that may be worth having in the standard like `love._os`, `love._version`, `love._version_codename`, `love._version_major`, `love._version_minor` and `love._version_revision` strings, and the `love.createhandlers` function

**Comment by giann** [comment]
Thanks for the review. @Positive07 should we remove things ? What of people using earlier version of Löve ?

**Comment by Positive07** [comment]
Staying with older versions is generally advised against... an advantage of luacheck is that you can always provide your own standard, and the 0.10.2 standard exists as a file already, which means you can always grab that standard file and use it. Alternatively you can add the missing features as an extra smaller standard, or just ignore those lines.

So yeah, moving the standard to 11.1 and removing anything that was removed is a sane decision.

On another note love-api remove functions that were removed, and instead has tagged releases for each LÖVE version (I could probably do the same on my luacheck-love repo)

**Comment by Positive07** [comment]
It looks great, I would call this ready to merge... What do you say about this stuff @giann shall it be added?
> There is also other "not so official" stuff that may be worth having in the standard like `love._os`, `love._version`, `love._version_codename`, `love._version_major`, `love._version_minor` and `love._version_revision` strings, and the `love.createhandlers` function

**Comment by giann** [comment]
Are those there since 11.0 or before ?

**Comment by mpeterv** [comment]
@giann thanks for the patch and @Positive07 thanks for the review! Feel free to open another PR if you decide to add these unofficial fields.

**Comment by Positive07** [comment]
@mpeterv No problem! Anytime, I'm fine with reviewing changes to the standard

@giann They are there since way way way before... `love._os` is what was used before `love.system.getOS` was implemented, and the version strings were used before `love.getVersion` came to be...

`love.createhandlers` is used (and declared) in `boot.lua` the file that starts the execution of a LÖVE program, it's a function that defines all the basic `love.handlers` and exists since `love.event` came to be.

**Pull request merged**